test/unit/test_http_client.py

```
@@ -0,0 +1,89 @@
+
+        debug_messages = [r.message for r in caplog.records if r.levelno == logging.DEBUG]
+        log_line = next((m for m in debug_messages if "Running urlopen" in m), None)
+        assert log_line is not None, "Expected 'Running urlopen' in debug log output"
+
+        assert "hf_secrettoken123" not in log_line
+        assert "****" in log_line
+
+    def test_authorization_header_still_sent_in_request(self):
+        client = HttpClient()
+        client.file_size = 0
+        token = "Bearer hf_secrettoken123"
```

Ruff (0.15.17)

[error] Possible hardcoded password assigned to: "token"

(S105)